tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/RetryConditionTest.java

```diff
@@ -215,4 +227,144 @@ public void testClassLevelRetryWithAbortOnTrue() {
         Assert.assertEquals(clientForClassLevelAbortOn.getRetryCountForWritingService(), 1,
             "The max invocation counter should be 1 as the abort condition is true");
     }
+
+
+    /**
+     * Persistent Error condition. Will retry 2 times and still throw exception.
+     * ServiceA uses {@link org.eclipse.microprofile.faulttolerance.Asynchronous} and will always return IOException.
+     */
+    @Test
+    public void testAsyncRetryExceptionally() {
+        final CompletionStage<String> future = asyncRetryClient.serviceA();
+
+        assertCompleteExceptionally(future, IOException.class, "Simulated error");
+        assertEquals(2, asyncRetryClient.getCountInvocationsServA());
+    }
+
+    /**
+     * Persistent Error condition inside a CompletableFuture.
+     * Will not retry because method is not marked with @Asynchronous
+     * ServiceB will always complete exceptionally with IOException.
+     */
+    @Test
+    public void testNoAsynWilNotRetryExceptionally() {
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        asyncRetryClient.serviceBFailExceptionally(future);
+
+        assertCompleteExceptionally(future, IOException.class, "Simulated error");
```

Comment:
Why does this assert on the value of `future` which was supplied as an argument, rather than the value returned from `serviceBFailExceptionally`?

Reply:
In this case it has the same outcome because the object reference is the same, but I agree the the returned value should be tested. Thanks.

Reply:
A few other test have a problem related to this. Issuing a fix